src/libraries/Common/tests/TestUtilities/System/Buffers/BoundedMemory.Windows.cs

```diff
@@ -288,38 +288,38 @@ public VirtualAllocHandle()
                 UnsafeNativeMethods.VirtualFree(handle, IntPtr.Zero, VirtualAllocAllocationType.MEM_RELEASE);
         }
 
-        private static partial class UnsafeNativeMethods
+        private static class UnsafeNativeMethods
```

Comment:
I had to change this to use the old way of defining p/invoke methods in C# because:
1. We cannot reference `TestUtilities.cspoj` in `Sve_ro.csproj`.
2. `Sve_ro.csproj` is not referencing the library import generator to properly handle `LibraryImport` attributes. I tried to reference it and it's not trivial.

This isn't a public API, it's only used internally so it's probably fine that we make this use the old p/invoke way.

Reply:
CC. @AaronRobinsonMSFT, @GrabYourPitchforks as an FYI.

Reply:
> I don't think directly including files across the high level directories is appropriate.

These are notably explicitly in `Common/tests/TestUtilities` because they are meant to be included/shared across projects.

> Can't we just redefine these declarations in the test project?

Not without duplicating the code and risking it getting out of sync over time, which is undesirable given this is testing infra specifically meant for validating security and other memory access correctness

Reply:
> These are notably explicitly in Common/tests/TestUtilities because they are meant to be included/shared across projects.

Within `src/libraries`, sure. Do we have any examples that cross from `src/tests` into `src/libraries`? From the sounds of it, it is more than just the P/Invokes, which is unfortunate.

Reply:
This looks like a much bigger oprganizational and engineering issue. If this isn't addressed properly we get forced design from the `src/tests`, which are runtime tests, into `src/libraries` - this is a clear inversion. The `src/libraries` source shouldn't be limited by how one writes runtime tests. Given how many of these APIs are spreading, it is highy probable more "utilities" will be shared and more style will be imposed in `src/libraries` from `src/tests`. Is there a reason we can't push these tests under `src/libraries`? Can we reduce the complexity in these tests and supplement them with more complex tests under `src/libraries`? The interop tests do just that. If one has native resources it goes under `src/tests`, but if it is purely managed, we try to put it under `src/libraries`.

Reply:
> Is there a reason we can't push these tests under src/libraries

The biggest issue is that libraries tests don't run under the same set of stress configurations nor as regularly and these being very lowlevel APIs do warrant the additional validation most of the time.

> The src/libraries source shouldn't be limited by how one writes runtime tests

I wouldn't really say its a limitation here. We're just directly using `DllImport` for something which is already almost blittable and only uses simply marshalling otherwise. It'd be trivial to make them properly blittable

Reply:
> The biggest issue is that libraries tests don't run under the same set of stress configurations nor as regularly and these being very lowlevel APIs do warrant the additional validation most of the time.

+1

> I wouldn't really say its a limitation here. 

It is concept we are introducing here that is the concern, not necessarily this instance. As we continue to rely and build upon source generators, it is likely this will expand. We should avoid pushing requirements up.

> We're just directly using DllImport for something which is already almost blittable and only uses simply marshalling otherwise. It'd be trivial to make them properly blittable

Great suggestion, please do that then and considered me signed off. Thanks for helping to find common ground.

Reply:
> It is concept we are introducing here that is the concern, not necessarily this instance. As we continue to rely and build upon source generators, it is likely this will expand. We should avoid pushing requirements up.

👍. I think it would be good to allow source generators to run here long term as well, its part of build not execution so there's nothing really that should fundamentally block it. We already have several generator like tools and have even migrated the runtime tests to using Xunit more recently, so source generators seems like goodness from there.

> Great suggestion, please do that then and considered me signed off. Thanks for helping to find common ground.

👍, same. Glad to better understand your concerns here as well

Reply:
@TIHan, I can push a commit to your branch to make these blittable if you'd like.

Otherwise you basically just want to change the existing signatures to wrapper APIs that call the actual P/Invokes with slightly different signatures.

For example:
```csharp
public static partial bool VirtualProtect(
    IntPtr lpAddress,
    IntPtr dwSize,
    VirtualAllocProtection flNewProtect,
    out VirtualAllocProtection lpflOldProtect)
{
    fixed (VirtualAllocProtection* oldProtect = &lpflOldProtect)
    {
        return VirtualProtect(lpAddress, dwSize, flNewProtect, oldProtect) != 0;
    }
}

[LibraryImport(KERNEL32_LIB, SetLastError = true)]
[return: MarshalAs(UnmanagedType.Bool)]
private static extern int VirtualProtect(IntPtr lpAddress, IntPtr dwSize, VirtualAllocProtection flNewProtect, VirtualAllocProtection* lpflOldProtect);
```

Basically you just need (in the `DllImport`:
* `bool` to be `int`, normalize outputs via `!= 0` and inputs via `x ? 1 : 0`
* `out T` and `ref T` to become a `T*`, pinning and passing down the pointer
* `VirtualAllocHandle` should be `IntPtr`, you'll need to expose a constructor that passes the handle down to the `SafeHandle` constructor and create a new instance on return
* `MEMORY_BASIC_INFORMATION` is already blittable and shouldn't need changes

Reply:
Sure, you can push a commit. 

Thanks @tannergooding and @AaronRobinsonMSFT for discussing the implications of having to do this.